// react-card-component/lib/index.js
import{css as r,jsx as o}from"@emotion/react";
/*! *****************************************************************************
Copyright (c) Microsoft Corporation.

Permission to use, copy, modify, and/or distribute this software for any
purpose with or without fee is hereby granted.

THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
PERFORMANCE OF THIS SOFTWARE.
***************************************************************************** */const e=r`
  width: 100%;
  height:100%;
  background:white;
  color:dimgray;
`,t=r`
  height:100%;
  position: relative;
  border-radius: inherit;
  overflow: hidden;
`,a=o=>o&&r`border-radius: 10px;`,n=o=>o&&r`cursor:pointer;`,s=o=>{return r`${e=o,JSON.stringify(e).replace(/"/g,"").replace(/((:{))/g,"{").replace(/,/g,";").slice(1,-1)}`;var e},l=r=>{switch(r){case"up":return b;case"down":return c;case"left":return p;case"right":return u;case"zoom":return g}},d=(o,e)=>o&&r`
  background: ${`rgba( 255, 255, 255, ${(null==e?void 0:e.transparency)||.2})`};
  backdrop-filter: ${`blur( ${(null==e?void 0:e.blur)||4}px )`};
  -webkit-backdrop-filter: ${`blur( ${(null==e?void 0:e.blur)||4}px )`};
  border: 1px solid rgba( 255, 255, 255, 0.18 );
`,i=r`transition: background 150ms ease-out, transform 150ms ease-out;`,b=r`
  ${i}
  :hover{
    border-color: #1EA7FD50;
    box-shadow: rgba(0, 0, 0, 0.06) 0 3px 10px 0;
    transform: translate3d(0, -4px, 0);
  }
`,c=r`
  ${i}

  :hover{
    border-color: #1EA7FD50;
    box-shadow: rgba(0, 0, 0, 0.06) 0 3px 10px 0;
    transform: translate3d(0, 4px, 0);
  }
`,p=r`
  ${i}
  :hover{
    border-color: #1EA7FD50;
    box-shadow: rgba(0, 0, 0, 0.06) 0 3px 10px 0;
    transform: translate3d(-4px, 0, 0);
  }
`,u=r`
  ${i}
  :hover{
    border-color: #1EA7FD50;
    box-shadow: rgba(0, 0, 0, 0.06) 0 3px 10px 0;
    transform: translate3d(4px, 0, 0);
  }
`,g=r`
  ${i}
  :hover{
    z-index:10;
    border-color: #1EA7FD50;
    box-shadow: rgba(0, 0, 0, 0.06) 0 3px 10px 0;
    transform: scale(1.1);
  }
`;export default i=>{var{children:b,background:c,outlined:p=!0,bordered:u=!1,style:g={},hoverType:x,glass:h=!1,glassOption:f={}}=i,v=function(r,o){var e={};for(var t in r)Object.prototype.hasOwnProperty.call(r,t)&&o.indexOf(t)<0&&(e[t]=r[t]);if(null!=r&&"function"==typeof Object.getOwnPropertySymbols){var a=0;for(t=Object.getOwnPropertySymbols(r);a<t.length;a++)o.indexOf(t[a])<0&&Object.prototype.propertyIsEnumerable.call(r,t[a])&&(e[t[a]]=r[t[a]])}return e}(i,["children","background","outlined","bordered","style","hoverType","glass","glassOption"]);const m=[e,(w=p,w&&r`border: 1px solid #00000010;`),a(u),(y=c,r`background:${y||"#f5f5f5"};`),n(!!(null==v?void 0:v.onClick)),l(x),s(g),d(h,f)];var y,w;return o("div",Object.assign({css:m},v),o("div",{css:[t]},b))};
